Glasgow class 6- Rahma Berhan - JS 2- week 3

RahmaB1 opened the pull request:
<!--

The title for your pull request should be made in this format

CITY CLASS_NO - FIRST_NAME LAST_NAME - MODULE - WEEK_NO

For example,

London Class 7 - Chris Owen - HTML/CSS - Week 1

Please complete the details below this message

-->

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

RahmaB1 committed:
done-ex:1

RahmaB1 committed:
ex:2 - not done yet

RahmaB1 committed:
done highlight-words

RahmaB1 committed:
.

RahmaB1 committed:
update to do list

RahmaB1 committed:
.

annacollins85 commented on 1-exercises/A-reading-list/script.js:
```diff
@@ -1,5 +1,27 @@
+    let pEl = document.createElement("p");
+    divEl.appendChild(pEl);
+
+    let ulEl = document.createElement("ul");
```
The issue with this is that you are creating a new list for each iteration. Instead you should create the unordered list outside the for loop so that there is only one, then you can append the list items to the same unordered list. Also, can we not just append the unordered list directly to "content"? It seems semantically a bit strange to have an unordered list inside a p tag. Apart from this it looks good!

annacollins85 commented on 1-exercises/B-highlight-words/script.js:
```diff
@@ -1,7 +1,42 @@
+        spanEl.style.backgroundColor = "";
+      }
+    });
+  }
```
Great! Nice work 👏

annacollins85 commented on 2-projects/1-todo-list/script.js:
```diff
@@ -1,6 +1,51 @@
+    // liBtn.innerText= "completed"
+    // liBtn.className = "btn btn-primary mb-3";
+
+    list.appendChild(taskDiv);
+    taskDiv.appendChild(taskEl);
```
I don't think we need `taskDiv` here. We can just append `taskEl` directly to the list. This way the list item is a direct child of the list. Semantically this is how we should use lists.

RahmaB1 replied:
Done. Thanks

annacollins85 commented on 2-projects/1-todo-list/script.js:
```diff
@@ -12,11 +57,29 @@ let todos = [
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
-function addNewTodo(event) {
-  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
+//not sure where best to locate the [ add to do ] event listener
```
You have added it correctly. It should be attached to the button 👏

annacollins85 reviewed (commented):
Good work! 👏 Keep it up!

Remember to remove unused comments from your code. Also, you have used `let` for everything but actually you could have just used const. It seems like the value might be changing when you add things to it, but in fact it just changes what's inside the variable, not the variable itself. If this is not clear let me know and we can go over it in class 🙂

RahmaB1 committed:
updates after review from Anna

RahmaB1 committed:
updates after review

RahmaB1 commented:
@annacollins85 Thank you for your review and feedback! I have addressed all the points you mentioned in my latest commits. Your guidance has been incredibly helpful. Appreciate your time and effort!
